src/ctia/task/migrate_es_stores.clj

```diff
@@ -0,0 +1,204 @@
+                     :_type mapping)
+             batch)]
+    (es-doc/bulk-create-doc
+     conn
+     prepared-docs
+     false)))
+
+(defn create-target-store
+  "create the target store, pushing its template"
+  [target-store]
+  (let [wildcard (str (:indexname target-store) "*")]
+    (log/warnf "purging index: %s"
+               wildcard)
+    (es-index/delete! (:conn target-store)
+                      wildcard))
+  (log/warnf "creating index template: %s"
+             (:indexname target-store))
+  (log/warnf "creating index: %s"
+             (:indexname target-store))
+
+  (es-index/create! (:conn target-store)
+                    (:indexname target-store)
+                    (:settings target-store)))
+
+(defn migrate-store
+  "migrate a single store"
+  [current-store
+   target-store
+   operations
+   batch-size
+   confirm?]
+
+  (when confirm?
+    (create-target-store target-store))
+  (let [store-size (-> (fetch-batch current-store 1 0)
+                       :paging
+                       :total-hits)]
+    (log/warnf "store size: %s records" store-size))
+  (loop [offset 0
+         migrated-count 0]
+    (let [{:keys [data paging]
+           :as batch}
+          (fetch-batch current-store
+                       batch-size
+                       offset)
+          next (:next paging)
+          offset (:offset next)
+          migrated (transduce operations conj data)
+          migrated-count (+ migrated-count
+                            (count migrated))]
+      (when (seq migrated)
+        (when confirm?
+          (store-batch target-store migrated))
+        (log/warnf "migrated: %s" migrated-count))
+      (if next
+        (recur offset migrated-count)
+        (log/warnf "finished migrating store: %s"
+                   (:indexname current-store))))))
+
+(defn migrate-store-indexes
+  "migrate all es store indexes"
+  [suffix ops batch-size confirm?]
+  (let [current-stores (stores->maps @stores)
+        target-stores
+        (source-store-maps->target-store-maps current-stores
+                                              suffix)
+        operations (compose-operations ops)
+        batch-size (or batch-size default-batch-size)]
+
+    (log/warnf "migrating stores: %s" (keys current-stores))
+    (log/warnf "batch size: %s" batch-size)
+
+    (doseq [[sk sr] current-stores]
+      (log/warnf "migrating store: %s" sk)
+      (migrate-store sr
+                     (sk target-stores)
+                     operations
+                     batch-size
+                     confirm?))))
+
+(defn -main
+  "invoke with lein run -m ctia.task.migrate-es-stores <suffix> <operations> <batch-size> <confirm?>"
+  [suffix ops batch-size confirm?]
+
+  (assert suffix
+          "Please provide an indexname suffix for target store creation")
+  (assert ops "Please provide a csv operation list")
+  (assert batch-size "Please specify a batch size")
+
+  (log/warn "migrating all ES Stores")
```

Comment:
Why use the warn log level everywhere? It's a manual operation, info should be sufficient.